app/src/main/java/com/benefit/activities/UserProfileActivity.java

```diff
@@ -1,91 +1,235 @@
+
+    }
+
+
     private void makeToast(String msg) {
         Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
     }
 
-    public void onClickEdit(View view) {
-        Toast.makeText(this, "Edit button was pressed", Toast.LENGTH_SHORT).show();
+
+    private void buildRecyclerView() {
+        // recycler itself
+        recyclerView = findViewById(R.id.items_recycler);
+        recyclerView.setHasFixedSize(true);
+        layoutManager = new GridLayoutManager(this, 2);
+        adapter = new ProductRecyclerAdapter(products);
+
+        recyclerView.setLayoutManager(layoutManager);
+        recyclerView.setAdapter(adapter);
+
+        adapter.setOnItemClickListener(new ProductRecyclerAdapter.OnItemClickListener() {
+            @Override
+            public void onItemClick(int position, View view) {
+                Product clickedProduct = (Product) view.getTag();
+                Intent intent = new Intent(getApplicationContext(), ProductPageActivity.class);
+                intent.putExtra("product", clickedProduct);
+                startActivity(intent);
+            }
+
+            @Override
+            public void onDeleteClick(int position, View deleteButtonView) {
+                Product productToDelete = getProductFromButtonView(deleteButtonView);
+                openDeleteDialog(productToDelete, position);
+            }
+
+            @Override
+            public void onEditClick(int position, View editButtonView) {
+                Product productToEdit = getProductFromButtonView(editButtonView);
+                Intent intent = new Intent(getApplicationContext(), GiveItemActivity.class);
+                intent.putExtra(getApplicationContext().getString(R.string.product_relay),
+                        productToEdit);
+                startActivity(intent);
+            }
+        });
     }
 
-    public void onClickChat(View view) {
-        Toast.makeText(this, "Chat button was pressed", Toast.LENGTH_SHORT).show();
+
+    private Product getProductFromButtonView(View view) {
+        View constraintLayout = (View) view.getParent();
+        View cardview = (View) constraintLayout.getParent();
+        return (Product) cardview.getTag();
     }
 
-    private void setHeaderListeners() {
-        HeaderClickListener.setHeaderListeners(this);
+    public void removeItem(int position) {
+        products.remove(position);
+        adapter.notifyItemRemoved(position);
     }
 
-    @Override
-    public void startActivity(Intent intent) {
-        intent.putExtra(getString(R.string.user_relay), user);
-        super.startActivity(intent);
+    private void openDeleteDialog(Product productToDelete, int position) {
+        long productId = productToDelete.getId();
+        String productTitle = productToDelete.getTitle();
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Are you sure you want to delete \"" + productTitle + "\" ?")
```

Comment:
Let's put strings that we present to the user in the strings file

Reply:
Alright, working on it.